### PR TITLE
RPC Framework

### PR DESCRIPTION
**Pull request by arschles**
this is a first cut for doing RPCs and wiring up a model all the way up to a route

TL;DR - overall I feel like there are too many moving pieces, especially when combined with #3

here's the stuff I'm thinking for the models and RPC:

- Creating a model was pretty easy. I just write up the struct and the server-side impl. I assumed that 
- I didn't wire up an event handler to the RPC but I think it'd be pretty easy to do that because `components.Todo` has a `TodoClient` in it, which can talk to the todo server
- The `TodoClient` could be where we add in the caching stuff later
- I didn't implement the RPC server but we should abstract it away with a single `models.StartRPCServer()` function that is auto-generated and you call it in the `main`
- there were tons of layers to wire up a model to UI. I did this:
    - Created a model & one server-side function for the model
    - assumed that an RPC server and typesafe client would be auto-generated (didn't implement that yet, but wrote a `models/todo_generated.go` to illustrate what it could look like
    - assumed that the server-side function would be wired up in the RPC server and the RPC server would be auto generated into a `StartRPCServer` function that you call in the `main`

then, here's what I noticed about wiring up the model to UI:

- I wanted a `components/todo_autogenerated.go` to just appear for me, and have that be wired all the way up to a `models.TodoClient`
- Then I had a bunch of cognitive load connecting the `components.Todo` up to the `Todo.html` component and then that up to the `routes/todo.html`
- ...I still don't get how that all works enough to make it fit together but I tried to make something that looks decent. Ideally the component and then the routes would be autogenerated anyway like Buffalo does
- There still isn't a client-side router yet. I have some thoughts that I'll put into another PR

### CODE REVIEW

File: factor/files/files.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package files
+
```
Comment (bketelsen) [inline factor/files/files.go]: nice move..

File: golden/components/todo_generated.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+// Permalink is a convenience function to get the permalink for this todo.
+func (t Todo) Permalink() string {
+	return fmt.Sprintf("/todos/%s", t.Get().ID.String())
+}
+
+func (t *Todo) Render() string {
+	tpl := template.Must(template.New("todo").Parse(TodoTemplate))
+	return tpl.Execute(
```
Comment (bketelsen) [inline golden/components/todo_generated.go]: this should be "return TodoTemplate"
It gets parsed in the markup package.
